discovery/manager.go

```diff
@@ -404,9 +411,26 @@ func (m *Manager) allGroups() map[string][]*targetgroup.Group {
 			n[pkey.setName] += len(tg.Targets)
 		}
 	}
+
+	// Send empty lists for subs without any targets to make sure old stale targets are dropped by consumers.
+	// See: https://github.com/prometheus/prometheus/issues/12858 for details.
```

Comment:
// maybe we can merge with the above loops?
    // The added blank data will not be always useful
    // This may slow up sending updates???